### PR TITLE
Add algolia search backend

### PR DESCRIPTION
**Pull request by jasonLaster**
This adds the logic for building the algolia search backend.

```JSON
{
  "hits": [
    {
      "name": "Energy Vault",
      "path": "energy-vault",
      "logo": "https://dl.airtable.com/.attachments/af118ad456098bf03256102113d683b0/c5f49aa3/011.jpeg",
      "sector": [
        "energy"
      ],
      "objectID": "energy-vault",
      "_snippetResult": {
        "name": {
          "value": "<em>Energy</em> Vault",
          "matchLevel": "full"
        },
        "path": {
          "value": "<em>energy</em>-vault",
          "matchLevel": "full"
        },
        "logo": {
          "value": "https://dl.airtable.com/.attachments/af118ad456098bf03256102113d683b0/c5f49aa3/011.jpeg",
          "matchLevel": "none"
        },
        "sector": [
          {
            "value": "<em>energy</em>",
            "matchLevel": "full"
          }
        ]
      },
      "_highlightResult": {
        "name": {
          "value": "<em>Energy</em> Vault",
          "matchLevel": "full",
          "fullyHighlighted": false,
          "matchedWords": [
            "energy"
          ]
        },
        "path": {
          "value": "<em>energy</em>-vault",
          "matchLevel": "full",
          "fullyHighlighted": false,
          "matchedWords": [
            "energy"
          ]
        },
        "logo": {
          "value": "https://dl.airtable.com/.attachments/af118ad456098bf03256102113d683b0/c5f49aa3/011.jpeg",
          "matchLevel": "none",
          "matchedWords": []
        },
        "sector": [
          {
            "value": "<em>energy</em>",
            "matchLevel": "full",
            "fullyHighlighted": true,
            "matchedWords": [
              "energy"
            ]
          }
        ]
      },
      "_rankingInfo": {
        "nbTypos": 0,
        "firstMatchedWord": 0,
        "proximityDistance": 0,
        "userScore": 286,
        "geoDistance": 0,
        "geoPrecision": 1,
        "nbExactWords": 1,
        "words": 1,
        "filters": 0
      }
    },
    {
      "name": "Energy Web",
      "path": "energy-web",
      "logo": "",
      "sector": "",
      "objectID": "energy-web",
      "_snippetResult": {
        "name": {
          "value": "<em>Energy</em> Web",
          "matchLevel": "full"
        },
        "path": {
          "value": "<em>energy</em>-web",
          "matchLevel": "full"
        },
        "logo": {
          "value": "",
          "matchLevel": "none"
        },
        "sector": {
          "value": "",
          "matchLevel": "none"
        }
      },
      "_highlightResult": {
        "name": {
          "value": "<em>Energy</em> Web",
          "matchLevel": "full",
          "fullyHighlighted": false,
          "matchedWords": [
            "energy"
          ]
        },
        "path": {
          "value": "<em>energy</em>-web",
          "matchLevel": "full",
          "fullyHighlighted": false,
          "matchedWords": [
            "energy"
          ]
        },
        "logo": {
          "value": "",
          "matchLevel": "none",
          "matchedWords": []
        },
        "sector": {
          "value": "",
          "matchLevel": "none",
          "matchedWords": []
        }
      },
      "_rankingInfo": {
        "nbTypos": 0,
        "firstMatchedWord": 0,
        "proximityDistance": 0,
        "userScore": 287,
        "geoDistance": 0,
        "geoPrecision": 1,
        "nbExactWords": 0,
        "words": 1,
        "filters": 0
      }
    },
    {
      "name": "Energy Storage Systems",
      "path": "energy-storage-systems",
      "logo": "",
      "sector": "",
      "objectID": "energy-storage-systems",
      "_snippetResult": {
        "name": {
          "value": "<em>Energy</em> Storage Systems",
          "matchLevel": "full"
        },
        "path": {
          "value": "<em>energy</em>-storage-systems",
          "matchLevel": "full"
        },
        "logo": {
          "value": "",
          "matchLevel": "none"
        },
        "sector": {
          "value": "",
          "matchLevel": "none"
        }
      },
      "_highlightResult": {
        "name": {
          "value": "<em>Energy</em> Storage Systems",
          "matchLevel": "full",
          "fullyHighlighted": false,
          "matchedWords": [
            "energy"
          ]
        },
        "path": {
          "value": "<em>energy</em>-storage-systems",
          "matchLevel": "full",
          "fullyHighlighted": false,
          "matchedWords": [
            "energy"
          ]
        },
        "logo": {
          "value": "",
          "matchLevel": "none",
          "matchedWords": []
        },
        "sector": {
          "value": "",
          "matchLevel": "none",
          "matchedWords": []
        }
      },
      "_rankingInfo": {
        "nbTypos": 0,
        "firstMatchedWord": 0,
        "proximityDistance": 0,
        "userScore": 285,
        "geoDistance": 0,
        "geoPrecision": 1,
        "nbExactWords": 0,
        "words": 1,
        "filters": 0
      }
    },
    {
      "name": "Energy Source LLC",
      "path": "energy-source-llc",
      "logo": "",
      "sector": "",
      "objectID": "energy-source-llc",
      "_snippetResult": {
        "name": {
          "value": "<em>Energy</em> Source LLC",
          "matchLevel": "full"
        },
        "path": {
          "value": "<em>energy</em>-source-llc",
          "matchLevel": "full"
        },
        "logo": {
          "value": "",
          "matchLevel": "none"
        },
        "sector": {
          "value": "",
          "matchLevel": "none"
        }
      },
      "_highlightResult": {
        "name": {
          "value": "<em>Energy</em> Source LLC",
          "matchLevel": "full",
          "fullyHighlighted": false,
          "matchedWords": [
            "energy"
          ]
        },
        "path": {
          "value": "<em>energy</em>-source-llc",
          "matchLevel": "full",
          "fullyHighlighted": false,
          "matchedWords": [
            "energy"
          ]
        },
        "logo": {
          "value": "",
          "matchLevel": "none",
          "matchedWords": []
        },
        "sector": {
          "value": "",
          "matchLevel": "none",
          "matchedWords": []
        }
      },
      "_rankingInfo": {
        "nbTypos": 0,
        "firstMatchedWord": 0,
        "proximityDistance": 0,
        "userScore": 284,
        "geoDistance": 0,
        "geoPrecision": 1,
        "nbExactWords": 0,
        "words": 1,
        "filters": 0
      }
    },
    {
      "name": "Energy Source",
      "path": "energy-source",
      "logo": "",
      "sector": "",
      "objectID": "energy-source",
      "_snippetResult": {
        "name": {
          "value": "<em>Energy</em> Source",
          "matchLevel": "full"
        },
        "path": {
          "value": "<em>energy</em>-source",
          "matchLevel": "full"
        },
        "logo": {
          "value": "",
          "matchLevel": "none"
        },
        "sector": {
          "value": "",
          "matchLevel": "none"
        }
      },
      "_highlightResult": {
        "name": {
          "value": "<em>Energy</em> Source",
          "matchLevel": "full",
          "fullyHighlighted": false,
          "matchedWords": [
            "energy"
          ]
        },
        "path": {
          "value": "<em>energy</em>-source",
          "matchLevel": "full",
          "fullyHighlighted": false,
          "matchedWords": [
            "energy"
          ]
        },
        "logo": {
          "value": "",
          "matchLevel": "none",
          "matchedWords": []
        },
        "sector": {
          "value": "",
          "matchLevel": "none",
          "matchedWords": []
        }
      },
      "_rankingInfo": {
        "nbTypos": 0,
        "firstMatchedWord": 0,
        "proximityDistance": 0,
        "userScore": 283,
        "geoDistance": 0,
        "geoPrecision": 1,
        "nbExactWords": 0,
        "words": 1,
        "filters": 0
      }
    },
    {
      "name": "Energy Renaissance",
      "path": "energy-renaissance",
      "logo": "",
      "sector": "",
      "objectID": "energy-renaissance",
      "_snippetResult": {
        "name": {
          "value": "<em>Energy</em> Renaissance",
          "matchLevel": "full"
        },
        "path": {
          "value": "<em>energy</em>-renaissance",
          "matchLevel": "full"
        },
        "logo": {
          "value": "",
          "matchLevel": "none"
        },
        "sector": {
          "value": "",
          "matchLevel": "none"
        }
      },
      "_highlightResult": {
        "name": {
          "value": "<em>Energy</em> Renaissance",
          "matchLevel": "full",
          "fullyHighlighted": false,
          "matchedWords": [
            "energy"
          ]
        },
        "path": {
          "value": "<em>energy</em>-renaissance",
          "matchLevel": "full",
          "fullyHighlighted": false,
          "matchedWords": [
            "energy"
          ]
        },
        "logo": {
          "value": "",
          "matchLevel": "none",
          "matchedWords": []
        },
        "sector": {
          "value": "",
          "matchLevel": "none",
          "matchedWords": []
        }
      },
      "_rankingInfo": {
        "nbTypos": 0,
        "firstMatchedWord": 0,
        "proximityDistance": 0,
        "userScore": 282,
        "geoDistance": 0,
        "geoPrecision": 1,
        "nbExactWords": 0,
        "words": 1,
        "filters": 0
      }
    },
    {
      "name": "Energy Impact Partners",
      "path": "energy-impact-partners",
      "logo": "https://dl.airtable.com/.attachments/1694f4c5ce3996a11229bcc5db586a89/281f5edf/logo-new-2.png",
      "sector": [
        "investors"
      ],
      "objectID": "energy-impact-partners",
      "_snippetResult": {
        "name": {
          "value": "<em>Energy</em> Impact Partners",
          "matchLevel": "full"
        },
        "path": {
          "value": "<em>energy</em>-impact-partners",
          "matchLevel": "full"
        },
        "logo": {
          "value": "https://dl.airtable.com/.attachments/1694f4c5ce3996a11229bcc5db586a89/281f5edf/logo-new-2.png",
          "matchLevel": "none"
        },
        "sector": [
          {
            "value": "investors",
            "matchLevel": "none"
          }
        ]
      },
      "_highlightResult": {
        "name": {
          "value": "<em>Energy</em> Impact Partners",
          "matchLevel": "full",
          "fullyHighlighted": false,
          "matchedWords": [
            "energy"
          ]
        },
        "path": {
          "value": "<em>energy</em>-impact-partners",
          "matchLevel": "full",
          "fullyHighlighted": false,
          "matchedWords": [
            "energy"
          ]
        },
        "logo": {
          "value": "https://dl.airtable.com/.attachments/1694f4c5ce3996a11229bcc5db586a89/281f5edf/logo-new-2.png",
          "matchLevel": "none",
          "matchedWords": []
        },
        "sector": [
          {
            "value": "investors",
            "matchLevel": "none",
            "matchedWords": []
          }
        ]
      },
      "_rankingInfo": {
        "nbTypos": 0,
        "firstMatchedWord": 0,
        "proximityDistance": 0,
        "userScore": 281,
        "geoDistance": 0,
        "geoPrecision": 1,
        "nbExactWords": 0,
        "words": 1,
        "filters": 0
      }
    },
    {
      "name": "Uprise Energy",
      "path": "uprise-energy",
      "logo": "",
      "sector": [
        "energy"
      ],
      "objectID": "uprise-energy",
      "_snippetResult": {
        "name": {
          "value": "Uprise <em>Energy</em>",
          "matchLevel": "full"
        },
        "path": {
          "value": "uprise-<em>energy</em>",
          "matchLevel": "full"
        },
        "logo": {
          "value": "",
          "matchLevel": "none"
        },
        "sector": [
          {
            "value": "<em>energy</em>",
            "matchLevel": "full"
          }
        ]
      },
      "_highlightResult": {
        "name": {
          "value": "Uprise <em>Energy</em>",
          "matchLevel": "full",
          "fullyHighlighted": false,
          "matchedWords": [
            "energy"
          ]
        },
        "path": {
          "value": "uprise-<em>energy</em>",
          "matchLevel": "full",
          "fullyHighlighted": false,
          "matchedWords": [
            "energy"
          ]
        },
        "logo": {
          "value": "",
          "matchLevel": "none",
          "matchedWords": []
        },
        "sector": [
          {
            "value": "<em>energy</em>",
            "matchLevel": "full",
            "fullyHighlighted": true,
            "matchedWords": [
              "energy"
            ]
          }
        ]
      },
      "_rankingInfo": {
        "nbTypos": 0,
        "firstMatchedWord": 1,
        "proximityDistance": 0,
        "userScore": 800,
        "geoDistance": 0,
        "geoPrecision": 1,
        "nbExactWords": 1,
        "words": 1,
        "filters": 0
      }
    },
    {
      "name": "Ubiquitous Energy",
      "path": "ubiquitous-energy",
      "logo": "",
      "sector": [
        "energy"
      ],
      "objectID": "ubiquitous-energy",
      "_snippetResult": {
        "name": {
          "value": "Ubiquitous <em>Energy</em>",
          "matchLevel": "full"
        },
        "path": {
          "value": "ubiquitous-<em>energy</em>",
          "matchLevel": "full"
        },
        "logo": {
          "value": "",
          "matchLevel": "none"
        },
        "sector": [
          {
            "value": "<em>energy</em>",
            "matchLevel": "full"
          }
        ]
      },
      "_highlightResult": {
        "name": {
          "value": "Ubiquitous <em>Energy</em>",
          "matchLevel": "full",
          "fullyHighlighted": false,
          "matchedWords": [
            "energy"
          ]
        },
        "path": {
          "value": "ubiquitous-<em>energy</em>",
          "matchLevel": "full",
          "fullyHighlighted": false,
          "matchedWords": [
            "energy"
          ]
        },
        "logo": {
          "value": "",
          "matchLevel": "none",
          "matchedWords": []
        },
        "sector": [
          {
            "value": "<em>energy</em>",
            "matchLevel": "full",
            "fullyHighlighted": true,
            "matchedWords": [
              "energy"
            ]
          }
        ]
      },
      "_rankingInfo": {
        "nbTypos": 0,
        "firstMatchedWord": 1,
        "proximityDistance": 0,
        "userScore": 792,
        "geoDistance": 0,
        "geoPrecision": 1,
        "nbExactWords": 1,
        "words": 1,
        "filters": 0
      }
    },
    {
      "name": "Sierra Energy",
      "path": "sierra-energy",
      "logo": "",
      "sector": [
        "energy"
      ],
      "objectID": "sierra-energy",
      "_snippetResult": {
        "name": {
          "value": "Sierra <em>Energy</em>",
          "matchLevel": "full"
        },
        "path": {
          "value": "sierra-<em>energy</em>",
          "matchLevel": "full"
        },
        "logo": {
          "value": "",
          "matchLevel": "none"
        },
        "sector": [
          {
            "value": "<em>energy</em>",
            "matchLevel": "full"
          }
        ]
      },
      "_highlightResult": {
        "name": {
          "value": "Sierra <em>Energy</em>",
          "matchLevel": "full",
          "fullyHighlighted": false,
          "matchedWords": [
            "energy"
          ]
        },
        "path": {
          "value": "sierra-<em>energy</em>",
          "matchLevel": "full",
          "fullyHighlighted": false,
          "matchedWords": [
            "energy"
          ]
        },
        "logo": {
          "value": "",
          "matchLevel": "none",
          "matchedWords": []
        },
        "sector": [
          {
            "value": "<em>energy</em>",
            "matchLevel": "full",
            "fullyHighlighted": true,
            "matchedWords": [
              "energy"
            ]
          }
        ]
      },
      "_rankingInfo": {
        "nbTypos": 0,
        "firstMatchedWord": 1,
        "proximityDistance": 0,
        "userScore": 680,
        "geoDistance": 0,
        "geoPrecision": 1,
        "nbExactWords": 1,
        "words": 1,
        "filters": 0
      }
    }
  ],
  "nbHits": 129,
  "page": 0,
  "nbPages": 13,
  "hitsPerPage": 10,
  "facets": {},
  "exhaustiveFacetsCount": true,
  "exhaustiveNbHits": true,
  "query": "energy",
  "params": "query=energy&hitsPerPage=10&page=0&analytics=false&attributesToRetrieve=*&highlightPreTag=%3Cais-highlight-0000000000%3E&highlightPostTag=%3C%2Fais-highlight-0000000000%3E&attributesToSnippet=*%3A20&getRankingInfo=true&snippetEllipsisText=%E2%80%A6&responseFields=*&enableABTest=false&facets=*%2C&tagFilters=",
  "index": "Pages",
  "serverUsed": "c9-use-1.algolia.net",
  "indexUsed": "Pages",
  "parsedQuery": "energy",
  "timeoutCounts": false,
  "timeoutHits": false,
  "processingTimeMS": 1
}
```

### CODE REVIEW

File: gatsby-config.js [hunk 1]
```diff
@@ -26,27 +28,41 @@ module.exports = {
...
-            mapping: { "Cover": `fileNode` },
-            tableLinks: [`Organizations`]
-          }
-        ]
-      }
+            mapping: { Cover: `fileNode` },
+            tableLinks: [`Organizations`],
+          },
+        ],
+      },
```
Comment (jasonLaster) [inline gatsby-config.js]: prettier stuff

File: gatsby-config.js [hunk 2]
```diff
@@ -26,27 +28,41 @@ module.exports = {
...
-      }
+      },
     },
     `gatsby-plugin-sharp`,
-    `gatsby-plugin-postcss`
+    `gatsby-plugin-postcss`,
   ],
 }
+
+if (process.env.ALGOLIA_ADMIN_KEY) {
```
Comment (jasonLaster) [inline gatsby-config.js]: There are 3 new algolia keys that are needed. I am not exactly sure how we'll share those internally for testing, but it should be doable

File: package.json [hunk 3]
```diff
@@ -23,7 +26,8 @@
     "prop-types": "^15.7.2",
     "react": "^16.12.0",
     "react-dom": "^16.12.0",
-    "react-helmet": "^5.2.1"
+    "react-helmet": "^5.2.1",
+    "react-instantsearch-dom": "^6.3.0"
```
Comment (jasonLaster) [inline package.json]: we're not using this yet, but we will be

**Comment by bloudermilk** [comment]
@jasonLaster I created an Algolia account and set up the variables on Netlify. Is there some configuration that needs to be done on the web dashboard, or is everything handled by your code?

I can share that login with you if you need to set anything up for production.

What do you think about devs creating their own accounts for local testing? Not sure what Algolia has built in for deadling with local environments.

**Comment by jasonLaster** [comment]
@bloudermilk everything is handled by the code. 

- the build uses production
- we could add the app id and search id in node.development

i like the idea of asking others to have an account if they want to work on updating the index. In terms of reading from algolia the app id and search id should be sufficient. We might want to just add these in our `.env.development` by default

**Pull request merged**